src/Domain/Service/NoticeAssembler.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Domain\Service;
+
+use App\DTO\Contribution;
+use App\Entity\Contributor;
+use App\Entity\DomainName;
+use App\Entity\MatchingContext;
+use App\Entity\Notice;
+use App\Helper\NoticeVisibility;
+use App\Repository\ContributorRepository;
+use App\Repository\DomainNameRepository;
+use InvalidArgumentException;
+
+class NoticeAssembler
+{
+    private $contributorRepository;
+    private $domainNameRepository;
+
+    public function __construct(ContributorRepository $contributorRepository, DomainNameRepository $domainNameRepository)
+    {
+        $this->contributorRepository = $contributorRepository;
+        $this->domainNameRepository = $domainNameRepository;
+    }
+
+    public function assemble(Contribution $contribution): Notice
+    {
+        if ($contribution->isAQuestion()) {
+            $questionedContributor = $this->contributorRepository->find($contribution->getToContributorId());
+
+            if (!$questionedContributor) {
+                throw new InvalidArgumentException("Questioned contributor doesn't exist.");
+            }
+        }
+
+        $contributor = self::assembleContributor($contribution);
+
+        // @todo /!\ Nothing here prevent a user to impersonate another one /!\
+        $existingContributor = $this->contributorRepository->findOneBy(['email' => $contributor->getEmail()]);
+        if ($existingContributor) {
```

Comment:
Hum, I realize from your comment the code is not doing what's supposed to do.
The `$questionedContributor` is not the `$contributor` but we don't make anything out of it.
Maybe it's not that important for now, I could add its name to the description though...

Reply:
Yes, that’s why in fact I didn’t just refactored but fixed (?) it, assuming that we want to have the target contributor as author, as it most likely will be the future author of this notice …

Right @MaartenLMEM @christpet ?

Reply:
:thinking: actually you might be right. 

But still, both of them must appears somewhere:
- The questioner contributor
- The questioned contributor

Reply:
And you’re right about that … I would suggest to put the questioner in the message, like : 

```
Question from <questioner> on <url> : <posted_message>
```

Maybe the url is superfluous though …

Reply:
> OMG trying this awseome feature, just created a **product team**  and we can now **@** them @dis-moi/product.
> Teams for everyone! :beers: 

How shall we handle question in the backoffice @dis-moi/product ?

Reply:
Nice ! Pierrick does have an account yet ?

Reply:
> How shall we handle question in the backoffice @dis-moi/product ?

As contributions manager, I would say : 
- Handle questionners the same way as contributors, but one condition : keep the notice private.  
So maybe the esaiest way is keep any question or contribution private in Back Office.

Reply:
@JalilArfaoui might be too abstract for @dis-moi/product team but I think as long we mention it somewhere (as suggested in the description) we should be fine for now.

About your suggestion I might turn it the other way around tho:
```
Question to <questionedContributor>: <posted_message>
```

Reply:
@lutangar Does it mean you want to set the questioner as notice's contributor ?

Reply:
@MaartenLMEM Yes, posted questions/contributions are private initially (in fact, @lutangar created 2 new status `DRAFT` and `QUESTION` to distinguish) but we won’t show them outside of the backoffice

My question to @dis-moi/product was : Should we also set any new self-created contributor account as disabled initially ? I was arguing that we can create it enabled, since the notices aren’t visible anyway. This way, once at least one notice has been validated and passed public, the contributor becomes automatically visible … No need to go enable it (which might be overlooked)

Reply:
And the other question to @dis-moi/product was :

When a question is posted :

- Should we set the **questionner** as notice’s contributor, and mention the **questionned** contributor in the message ? 

It seems to me that @lutangar would opt for this one

- Should we set the **questionned** as notice’s contributor, and mention the **questionner** email/name in the message ?

I prefer this one, because the questionned contributor is likely to stay the contributor of the notice, once the reponse has been provided to the question. I see the `QUESTION` state as a really premature notice state, before  `DRAFT` and `PUBLIC` …

Moreover, choosing the first solution would lead to creating contributor accounts for people who may never contribute, but just post questions …

Reply:
@JalilArfaoui this discussion is going way too much into details for @dis-moi/product  at the moment... I'll implements your proposal, either way are fine by me.